src/transformers/trainer.py

```diff
@@ -417,13 +420,20 @@ def __init__(
             elif FSDPOption.NO_SHARD in args.fsdp:
                 self.fsdp = ShardingStrategy.NO_SHARD
 
-            self.backward_prefetch = BackwardPrefetch.BACKWARD_PRE
+            if isinstance(self.args.fsdp_config, dict):
+                self.fsdp_config = self.args.fsdp_config
+            elif isinstance(self.args.fsdp_config, str):
+                with io.open(self.args.fsdp_config, "r", encoding="utf-8") as f:
+                    self.fsdp_config = json.load(f)
+            else:
+                self.fsdp_config = {}
```

Comment:
This is super nice, but should be done in the post init of the TrainingArguments :-)